File: SMI-SDK/.NET SDK/Library/gis/trace/LinkedEntry.cs
namespace pt.sapo.gis.trace
{
    using System;
    using System.Collections.Generic;
    using pt.sapo.gis.trace.configuration;

    /**
     * @author GIS Team
     * Class for entry with an linked trace Id
     */
    public class LinkedEntry : Entry
    {
        public static readonly String LINKED_TRACE_ID_PROPERTY = "linkedTraceId";

        public String LinkedTraceId
        {
            get
            {
                return (String)Properties[LINKED_TRACE_ID_PROPERTY];
            }
            set
            {
                Properties.Add(LINKED_TRACE_ID_PROPERTY, value);
            }
        }

        public LinkedEntry() { }

        public LinkedEntry(String description, severityType severity) : base(description, severity) { }

        public LinkedEntry(String description, severityType severity, IDictionary<String, Object> properties) : base(description, severity, properties) { }
    }
}

File: SMI-SDK/Java SDK/src/pt/sapo/gis/trace/LinkedEntry.java
package pt.sapo.gis.trace;

import java.util.Map;
import pt.sapo.gis.trace.configuration.SeverityType;

/**
 * @author GIS Team
 * Class for entry with an linked trace Id
 */
public class LinkedEntry extends Entry{
    public static final String LINKED_TRACE_ID_PROPERTY = "linkedTraceId";
    
    public String getLinkedTraceId() {        
        return (String) getProperty(LINKED_TRACE_ID_PROPERTY);
    }
    
    public final void setLinkedTraceId(String linkedTraceId) {
        setProperty(LINKED_TRACE_ID_PROPERTY, linkedTraceId);
    }
    
    public LinkedEntry(){
        super();
    }
    
    public LinkedEntry(String description, SeverityType severity){    
        super(description, severity);
    }
    
    public LinkedEntry(String description, SeverityType severity, Map<String,Object> properties){    
        super(description, severity, properties);
    }
}